android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java

```diff
@@ -211,6 +212,52 @@ public void clear() {
         }
     }
 
+    @Override
+    public <T extends GenericJson> void clearCollection(String collection, Class<T> collectionItemClass, Long ttl) {
+        synchronized (LOCK) {
+            DynamicRealm realm = getDynamicRealm();
+            try {
+                realm.beginTransaction();
+                List<String> schemas = new ArrayList<>();
+                schemas.add(TableNameManager.getShortName(collection, realm));
+                schemas.addAll(getEmbeddedObjectSchemas(collection, realm));
+                removeSchemas(schemas, realm);
+                String cacheKey = getClientHash() + File.separator + collection;
+                RealmCache<T> cache = (RealmCache<T>) mCacheMap.get(cacheKey);
+                if (cache != null) {
+                    cache.createRealmTable(realm);
+                }
+                realm.commitTransaction();
+            } finally {
+                realm.close();
+            }
+        }
+    }
+
+    private List<String> getEmbeddedObjectSchemas(String tableName, DynamicRealm realm) {
+        RealmSchema currentSchema = realm.getSchema();
+        String originalName;
+        String className;
+        List<String> embeddedObjectSchemas = new ArrayList<>();
+        Set<RealmObjectSchema> schemas = currentSchema.getAll();
+        for (RealmObjectSchema schema : schemas) {
+            className = schema.getClassName();
+            if (className.equals(TableNameManager.getShortName(tableName, realm))) {
+                for (RealmObjectSchema subClassSchema : schemas) {
+                    originalName = TableNameManager.getOriginalName(subClassSchema.getClassName(), realm);
+                    if (originalName == null) {
+                        continue;
+                    }
+                    if (originalName.startsWith(className + "_")) {
```

Comment:
Should use the underscore constant here if possible.

Reply:
I agree. Constants.UNDERSCORE was added to another branch (MLIBZ-2392). I'll update it after merging MLIBZ-2392 branch to indev.